### PR TITLE
feat: only cache if the SDK is not yet cached

### PR DESCRIPTION
**Pull request by quad**

### CODE REVIEW

**Comment by amyu** [comment]
please run `npm run all` and commit!

**Comment by amyu** [comment]
and check the lint error(prettier)

File: src/cache.ts [hunk 1]
```diff
@@ -56,6 +56,13 @@ export async function saveCache(
     cacheKey
   )
 
+  core.info(`checking if "${restoreKey}" is already cached ...`)
+  const hasEntry = await cache.restoreCache([ANDROID_HOME_DIR], restoreKey, [], { lookupOnly: true })
+  if (hasEntry) {
+    core.info(`Found in cache: ${restoreKey}`)
+    return
+  }
+
```
[the remark this answers is not in the document]
Comment (quad) [reply src/cache.ts:65]: Are you suggesting that I remove the catch on `ReserveCacheError`? 👌

[the remark this answers is not in the document]
Comment (amyu) [reply src/cache.ts]: Yes!
I did not know how to check if the cache already exists.
So, if I did a cache and got a ReserveCacheError, it meant that the cache was already there.

**Comment by amyu** [comment]
Thanks for your PR!

**Comment by amyu** [comment]
https://github.com/amyu/setup-android/releases/tag/v3.3
Released in v3.3

**Comment by quad** [comment]
Thank you!

**Pull request merged**